chrome/content/zotero/xpcom/itemTreeManager.js

```diff
@@ -180,40 +182,43 @@ class ItemTreeManager {
 
 	/**
 	 * Add a new column or new columns. All options must be valid. An atomic operation.
-	 * @param {ItemTreeColumnOption | ItemTreeColumnOption[]} optionOrOptions - An option or array of options to add
+	 * @param {ItemTreeColumnOption | ItemTreeColumnOption[]} options - An option or array of options to add
+	 * @param {boolean} [skipValidate=false] - Whether to validate the option(s) before adding 
 	 * @returns {boolean} - True if column(s) were added, false if not
 	 */
-	_addColumn(optionOrOptions) {
+	_addColumns(options, skipValidate) {
 		// If any check fails, return check results
-		if (!this._validateColumnOption(optionOrOptions)) {
+		if (!skipValidate && !this._validateColumnOption(options)) {
```

Comment:
This is a confusing way to write this. Can we not just check for invalid data keys on the array, and then perform whatever operations for individual column option validation later?

Reply:
It's bad because this code has weird execution flow based on both arguments and use of recursion. Recursion is generally reserved for tree data structures.

If a function can receive both a single item, or an array of items, the convention is to convert the single item into an array of length 1 and then the rest of the function treats the argument as an array. That way you can read the function from top to bottom and see the code flow.

So at the top you would do
```
if (!Array.isArray(options)) {
  options = [options]
}
```
And simplify the rest of the function.

Reply:
Done